diff -Naur nessus-core/Makefile nessus-core-p/Makefile
--- nessus-core/Makefile	Tue May 28 10:09:38 2002
+++ nessus-core-p/Makefile	Sun Sep  8 21:43:17 2002
@@ -20,51 +20,51 @@
 	@echo
 
 install-bin:
-	test -d ${sbindir} || $(INSTALL_DIR) -m 755 ${sbindir}
-	$(INSTALL) -m $(SERVERMODE) -o $(installuser) \
-		${make_bindir}/nessusd ${sbindir}
-	test -d ${sysconfdir} || $(INSTALL_DIR) -m 755 ${sysconfdir}
-	test -d ${sysconfdir}/nessus || $(INSTALL_DIR) -m 755 ${sysconfdir}/nessus	
-	test -d ${NESSUSD_DATADIR} || \
-		$(INSTALL_DIR) -m $(PLUGINSDIRMODE) ${NESSUSD_DATADIR}
-	test -d $(NESSUSD_PLUGINS) || \
-		$(INSTALL_DIR) -m $(PLUGINSDIRMODE) $(NESSUSD_PLUGINS)
-	test -d ${includedir} || $(INSTALL_DIR) -m 755 ${includedir}
-	test -d ${includedir}/nessus || $(INSTALL_DIR) -m 755 ${includedir}/nessus
-	test -d ${localstatedir} || $(INSTALL_DIR) -m 755 ${localstatedir}
-	test -d ${NESSUSD_STATEDIR} || $(INSTALL_DIR) -m 755 ${NESSUSD_STATEDIR}
-	test -d ${NESSUSD_STATEDIR}/users || $(INSTALL_DIR) -m 755 ${NESSUSD_STATEDIR}/users
-	test -d ${NESSUSD_STATEDIR}/logs  || $(INSTALL_DIR) -m 755 ${NESSUSD_STATEDIR}/logs
-	test -d ${NESSUSD_STATEDIR}/jobs  || $(INSTALL_DIR) -m 755 ${NESSUSD_STATEDIR}/jobs
-	$(INSTALL) -c -m 0444 include/config.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/ntcompat.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/includes.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/nessus-devel.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/nessusraw.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/nessusip.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/nessusicmp.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/nessustcp.h ${includedir}/nessus
-	$(INSTALL) -c -m 0444 include/nessusudp.h ${includedir}/nessus
-	$(INSTALL) -m $(CLIENTMODE) nessus-adduser ${sbindir}
-	$(INSTALL) -m $(CLIENTMODE) nessus-rmuser ${sbindir}
-	$(INSTALL) -m $(CLIENTMODE) nessus-mkcert ${sbindir}
-	$(INSTALL) -m $(CLIENTMODE) nessus-mkcert-client ${bindir}
-	$(INSTALL) -m $(CLIENTMODE) ssl/nessus-mkrand ${bindir}
+	test -d $(DESTDIR)${sbindir} || $(INSTALL_DIR) -m 755 $(DESTDIR)${sbindir}
+	$(INSTALL) -m $(SERVERMODE) \
+		${make_bindir}/nessusd $(DESTDIR)${sbindir}
+	test -d $(DESTDIR)${sysconfdir} || $(INSTALL_DIR) -m 755 $(DESTDIR)${sysconfdir}
+	test -d $(DESTDIR)${sysconfdir}/nessus || $(INSTALL_DIR) -m 755 $(DESTDIR)${sysconfdir}/nessus	
+	test -d $(DESTDIR)${NESSUSD_DATADIR} || \
+		$(INSTALL_DIR) -m $(PLUGINSDIRMODE) $(DESTDIR)${NESSUSD_DATADIR}
+	test -d $(DESTDIR)$(NESSUSD_PLUGINS) || \
+		$(INSTALL_DIR) -m $(PLUGINSDIRMODE) $(DESTDIR)$(NESSUSD_PLUGINS)
+	test -d $(DESTDIR)${includedir} || $(INSTALL_DIR) -m 755 $(DESTDIR)${includedir}
+	test -d $(DESTDIR)${includedir}/nessus || $(INSTALL_DIR) -m 755 $(DESTDIR)${includedir}/nessus
+	test -d $(DESTDIR)${localstatedir} || $(INSTALL_DIR) -m 755 $(DESTDIR)${localstatedir}
+	test -d $(DESTDIR)${NESSUSD_STATEDIR} || $(INSTALL_DIR) -m 755 $(DESTDIR)${NESSUSD_STATEDIR}
+	test -d $(DESTDIR)${NESSUSD_STATEDIR}/users || $(INSTALL_DIR) -m 755 $(DESTDIR)${NESSUSD_STATEDIR}/users
+	test -d $(DESTDIR)${NESSUSD_STATEDIR}/logs  || $(INSTALL_DIR) -m 755 $(DESTDIR)${NESSUSD_STATEDIR}/logs
+	test -d $(DESTDIR)${NESSUSD_STATEDIR}/jobs  || $(INSTALL_DIR) -m 755 $(DESTDIR)${NESSUSD_STATEDIR}/jobs
+	$(INSTALL) -c -m 0444 include/config.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/ntcompat.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/includes.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/nessus-devel.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/nessusraw.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/nessusip.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/nessusicmp.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/nessustcp.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -c -m 0444 include/nessusudp.h $(DESTDIR)${includedir}/nessus
+	$(INSTALL) -m $(CLIENTMODE) nessus-adduser $(DESTDIR)${sbindir}
+	$(INSTALL) -m $(CLIENTMODE) nessus-rmuser $(DESTDIR)${sbindir}
+	$(INSTALL) -m $(CLIENTMODE) nessus-mkcert $(DESTDIR)${sbindir}
+	$(INSTALL) -m $(CLIENTMODE) nessus-mkcert-client $(DESTDIR)${bindir}
+	$(INSTALL) -m $(CLIENTMODE) ssl/nessus-mkrand $(DESTDIR)${bindir}
 
 
 install-man:
 	@echo installing man pages ...
-	@test -d ${mandir}/man1 || $(INSTALL_DIR) ${mandir}/man1
-	@test -d ${mandir}/man8 || $(INSTALL_DIR) ${mandir}/man8
+	@test -d $(DESTDIR)${mandir}/man1 || $(INSTALL_DIR) $(DESTDIR)${mandir}/man1
+	@test -d $(DESTDIR)${mandir}/man8 || $(INSTALL_DIR) $(DESTDIR)${mandir}/man8
 
-	$(INSTALL) -c -m 0444 ${MAN_NESSUS_1} ${mandir}/man1/nessus.1
-	$(INSTALL) -c -m 0444 ${MAN_NESSUSD_8} ${mandir}/man8/nessusd.8
-	$(INSTALL) -c -m 0444 doc/nessus-adduser.8 ${mandir}/man8/nessus-adduser.8
-	$(INSTALL) -c -m 0444 doc/nessus-rmuser.8 ${mandir}/man8/nessus-rmuser.8
-	$(INSTALL) -c -m 0444 doc/nessus-mkcert.8 ${mandir}/man8/nessus-mkcert.8
+	$(INSTALL) -c -m 0444 ${MAN_NESSUS_1} $(DESTDIR)${mandir}/man1/nessus.1
+	$(INSTALL) -c -m 0444 ${MAN_NESSUSD_8} $(DESTDIR)${mandir}/man8/nessusd.8
+	$(INSTALL) -c -m 0444 doc/nessus-adduser.8 $(DESTDIR)${mandir}/man8/nessus-adduser.8
+	$(INSTALL) -c -m 0444 doc/nessus-rmuser.8 $(DESTDIR)${mandir}/man8/nessus-rmuser.8
+	$(INSTALL) -c -m 0444 doc/nessus-mkcert.8 $(DESTDIR)${mandir}/man8/nessus-mkcert.8
 	$(INSTALL) -c -m 0444 doc/nessus-mkcert-client.1 \
-                              ${mandir}/man1/nessus-mkcert-client.1
-	$(INSTALL) -c -m 0444 doc/nessus-mkrand.1 ${mandir}/man1/nessus-mkrand.1
+                              $(DESTDIR)${mandir}/man1/nessus-mkcert-client.1
+	$(INSTALL) -c -m 0444 doc/nessus-mkrand.1 $(DESTDIR)${mandir}/man1/nessus-mkrand.1
 
 win32: ${MAN_NESSUS_1} ${MAN_NESSUSD_8}
 	$(MANROFF) ${MAN_NESSUS_1}  > doc/nessus.1.cat
@@ -77,8 +77,8 @@
 	@echo
 
 client-install : client
-	test -d ${bindir} || $(INSTALL_DIR) -m 755 ${bindir}
-	$(INSTALL) -m $(CLIENTMODE) ${make_bindir}/nessus ${bindir}
+	test -d $(DESTDIR)${bindir} || $(INSTALL_DIR) -m 755 $(DESTDIR)${bindir}
+	$(INSTALL) -m $(CLIENTMODE) ${make_bindir}/nessus $(DESTDIR)${bindir}
 
 client : 
 	cd nessus && $(MAKE)
